sklearnex/cluster/tests/test_kmeans.py

```diff
@@ -128,8 +128,10 @@ def test_results_on_dense_gold_data(dataframe, queue, algorithm):
 def test_dense_vs_sparse(queue, init, algorithm, dims):
     from sklearnex.cluster import KMeans
 
-    if init == "random":
-        pytest.skip("Random initialization in sparse K-means is buggy.")
+    if init in ["random", "k-means++"]:
+        pytest.skip(
+            "Random and K-means++ initialization for sparse K-means are non-conformant."
+        )

```

Comment:
If this reproduced only github actions? I guess we can you _IS_INTEL for skip condition